plugin/src/main/scala/NGPlugin.scala

```diff
@@ -20,17 +20,43 @@ import de.johoop.testngplugin.TestNGPlugin._
 
 object NGPlugin extends Plugin {
 
+  @deprecated("Use ngProjectSettings or ngBuildSettings instead", "2.5.0")
   def ngSettings: Seq[Setting[_]] = super.settings ++ Seq(
     testOptions := Seq(),
     //testOptions += Tests.Argument(TestFrameworks.Specs2, "sequential", "true"),
-    testOptions += Tests.Argument(TestFrameworks.JUnit,"junitxml", "console")
-   ) ++
-   testNGSettings ++
-   Seq(
+    testOptions += Tests.Argument(TestFrameworks.JUnit,"junitxml", "console")) ++
+    testNGSettings ++
+    Seq(
      testNGParameters ++= Seq("-listener", "com.linkedin.plugin.FailSkippedTestsListener"),
-       libraryDependencies <++= (testNGVersion)(v => Seq(
+       libraryDependencies ++= testNGVersion(v => Seq(
          "org.testng" % "testng" % v % "test->default",
          // If changing this, be sure to change in Build.scala also.
-         "de.johoop" %% "sbt-testng-interface" % "3.0.2" % "test"))
+         "de.johoop" %% "sbt-testng-interface" % "3.0.2" % "test")).value
   )
+
+  def ngProjectSettings: Seq[Setting[_]] = super.projectSettings ++ Seq(
```

Comment:
Please see [here](http://www.scala-sbt.org/0.13/docs/Plugins.html#projectSettings+and+buildSettings) for the difference between projectSettings and buildingSettings. We don't need to copy everything.

Reply:
Done.